src/core/mesh/qgsmeshrenderersettings.h

```diff
@@ -315,6 +278,187 @@ class CORE_EXPORT QgsMeshRendererVectorSettings
+      //!Displying vctor dataset with arrows
+      Arrows = 0,
+      //!Displying vctor dataset with streamlines
+      Streamlines
+    };
+
+
+    //! Returns line width of the arrow (in millimeters)
+    double lineWidth() const;
+    //! Sets line width of the arrow in pixels (in millimeters)
+    void setLineWidth( double lineWidth );
+
+    //! Returns color used for drawing arrows
+    QColor color() const;
+    //! Sets color used for drawing arrows
+    void setColor( const QColor &color );
+
+    /**
+     * Returns filter value for vector magnitudes.
+     *
+     * If magnitude of the vector is lower than this value, the vector is not
+     * drawn. -1 represents that filtering is not active.
+     */
+    double filterMin() const;
+
+    /**
+     * Sets filter value for vector magnitudes.
+     * \see filterMin()
+     */
+    void setFilterMin( double filterMin );
+
+    /**
+     * Returns filter value for vector magnitudes.
+     *
+     * If magnitude of the vector is higher than this value, the vector is not
+     * drawn. -1 represents that filtering is not active.
+     */
+    double filterMax() const;
+
+    /**
+     * Sets filter value for vector magnitudes.
+     * \see filterMax()
+     */
+    void setFilterMax( double filterMax );
+
+    //! Returns the displaying method used to render vector datasets
+    Symbology displayingMethod() const;
+    //! Sets the displaying method used to render vector datasets
+    void setDisplayingMethod( const Symbology &displayingMethod );
+
+    //! Returns settings for vector rendered with arrows
+    QgsMeshRendererVectorArrowSettings arrowSettings() const;
+    void setArrowsSettings( const QgsMeshRendererVectorArrowSettings &arrowSettings );
+
+    //! Returns settings for vector rednered with streamlines
+    QgsMeshRendererVectorStreamlineSettings streamLinesSettings() const;
+    void setStreamLinesSettings( const QgsMeshRendererVectorStreamlineSettings &streamLinesSettings );
+
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc ) const
+    {
+      QDomElement elem = doc.createElement( QStringLiteral( "vector-settings" ) );
+      elem.setAttribute( QStringLiteral( "symbology" ), mDisplayingMethod );
+
+      elem.setAttribute( QStringLiteral( "line-width" ), mLineWidth );
+      elem.setAttribute( QStringLiteral( "color" ), QgsSymbolLayerUtils::encodeColor( mColor ) );
+      elem.setAttribute( QStringLiteral( "filter-min" ), mFilterMin );
+      elem.setAttribute( QStringLiteral( "filter-max" ), mFilterMax );
+
+      elem.appendChild( mArrowsSettings.writeXml( doc ) );
+      elem.appendChild( mStreamLinesSettings.writeXml( doc ) );
+
+      return elem;
+
+    }
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem )
+    {
+      mArrowsSettings.readXml( elem );
+      //Old version (<3.12). If new version, return empty settings and the real settings will be read bellow
```

Comment:
i do not understand this comment. does it belong to mArrowsSettings.readXML? maybe put it in that function. usually we have "TODO: remove in QGIS4" . this whole API is experimental, so I assume we do not need to be super-bacward compatible.